### ISSUE TITLE
Recipes for several machines are using the water tag when they shouldn't

### ISSUE THREAD

**Issue by jamesjimmy88** [mekanism/Mekanism#7159]
#### Issue description:
The recipes that require water are using the tag "minecraft:water", which is also used by the game to make tagged fluids behave like water, which also means that the tag will be used by several other mods for their fluids, like for example Create's Liquid Honey and Chocolate. 
The devices involved are several (Chemical Washer, Electrolytic Separator, PRC, Rotary Condensentrator, Thermal Evaporation Tower and Fission Reactor), but I guess it's just a matter of replacing all the occurrences of _"tag":"minecraft:water"_ with _"fluid":"minecraft:water"_ in the _data/recipes/_ section of Mekanism/Generators' code.

#### Steps to reproduce:
1. Make some chocolate with Create
2. Pump it into an Electrolytic Separator
3. Get Hydrogen and Oxygen

#### Version (make sure you are on the latest version before reporting):
**Forge:** 36.1.13
**Mekanism:** 10.0.21.448

**Comment by Robball14** [comment]
this is on create's end, not ours
P.S.
try cooling a reactor using chocolate instead of water for a bit of extra challenge

**Comment by pupnewfster** [comment]
Report this to Create. It is a bug on their end that they are declaring honey/chocolate to be water when they aren't as the water tag is used to declare *a lot* of different mechanics that specifically water has so if they are tagging it as water that means it is equivalent to water. There are some people in the process of actually doing the *correct* thing that create should either help with/tried to do if they actually care about fluid interactions and not just using a massive "hack" to get their fluids to have interactions, which is to make a PR to forge decoupling the corresponding vanilla logic from the water tag.

**Comment by jamesjimmy88** [comment]
Will do, thanks

**Comment by SpottyTheTurtle** [comment]
Not a bug, it's tagged as water because they wanted players to be able to swim, ride boats on, etc on honey and chocolate in-world. This is just one of a few side effects caused by this choice.

**Comment by fengshuo2004** [comment]
> Report this to Create. It is a bug on their end that they are declaring honey/chocolate to be water when they aren't as the water tag is used to declare _a lot_ of different mechanics that specifically water has so if they are tagging it as water that means it is equivalent to water. There are some people in the process of actually doing the _correct_ thing that create should either help with/tried to do if they actually care about fluid interactions and not just using a massive "hack" to get their fluids to have interactions, which is to make a PR to forge decoupling the corresponding vanilla logic from the water tag.

This is hardly a bug of Create. Tagging fluids `minecraft:water` to give them entity interaction is common practice for mods 1.18.2 and below. It was impossible for a modded fluid to inherit the entity interaction of vanilla water (I think) until the 1.19 Forge API fixed this by introducing the `FluidType` class.

TL;DR this can be fixed by a) _Telling every other mod dev to not tag their fluid `minecraft:water` and that they'd write a big chunk of code re-implementing entity interactions_ OR b) _Changing Mekanism's recipe to use the fluid id not the tag_. I think it's clear which is the easier solution.
